### ISSUE TITLE
Null pointer exception and reloading

### ISSUE THREAD

**Issue by httpnotonly** [Timmotools/flipperzero_esp8266_deautherv2#5]
The faced behavior is in title actually :)

**Comment by Timmotools** [comment]
When are you getting this error? You have to wait a few seconds before
sending any commands. Check log end to confirm no messages are being sent
from the esp board. Other thing to check is the firmware in the esp8266.
This will only work if you are using the esp8266 deauther v2 firmware.

**Comment by httpnotonly** [comment]
https://user-images.githubusercontent.com/38507289/207261692-d5896229-f8de-4502-a6cb-63f960e3d476.MOV


https://user-images.githubusercontent.com/38507289/207261720-bb80fd85-9c53-4cb6-8842-b87a39f30c21.MOV

**Comment by Timmotools** [comment]
Are you getting any data if you read the log? (First option).

**Comment by httpnotonly** [comment]
there are .. some :)
The most interesting part is that other options are replaced with these logs when I visit them first.
If I click on something else - null pointer

https://user-images.githubusercontent.com/38507289/207270319-df984eed-1f7d-4d13-a07c-6884c4d2c758.MOV

**Comment by httpnotonly** [comment]
Thanks for your rapid replies btw :)

**Comment by Timmotools** [comment]
It looks like you don't have the right firmware loaded on the esp.

Check you have this one https://github.com/SpacehuhnTech/esp8266_deauther.

**Comment by httpnotonly** [comment]
I had
+have recorded another flashing process

https://user-images.githubusercontent.com/38507289/207273949-34fa5ca2-bbae-4c4f-8fe2-a570cf95a209.mov

**Comment by httpnotonly** [comment]
Unleashed 18
(with the latest 20 it's not working)

**Comment by Timmotools** [comment]
That's really strange. I did notice that the esp was still plugged into the
usb. I know some of them require a reset or power cycle after programming.
This is so the esp leaves flash mode and boots the flashed code.

I'm just going to put my daughter to bed and then I'll grab one of my
deauther boards from the car to test out the latest build of unleashed.
I'll have to build unleashed from source as I run some custom libraries.

The other option is trying the flipper firmware I have on another repo. Nfc
gamecard hack. That has the deauther on the main menu and extra apps. I
know that one worked. Unleashed with an NFC change to generate random game
cards in Melbourne. I have covid so I can't test that one just yet.

You can also make sure that the uart is working by going to gpio on the
flipper and use the uart bridge on port 1. Then use your favourite terminal
application with baud 115200. I use putty because I'm a Windows n00b lol.

**Comment by httpnotonly** [comment]
>That's really strange. I did notice that the esp was still plugged into the
usb. I know some of them require a reset or power cycle after programming.
This is so the esp leaves flash mode and boots the flashed code.

I have just ejected it from laptop to not to damage my small installation :)

Take your time. Anyway I did not see such a fast support in open-source for a long time haha

**Comment by Timmotools** [comment]
Did you happen to have a look at the flashing video I did a week back?
Might be some useful info there.

I apologise for the audio quality. I didn't realise it was that bad until
I'd finished recording and started editing it.

I did just confirm the version in my other repo does work with my esp
board. But that's an esp8266ex board.

**Comment by Timmotools** [comment]
https://youtu.be/CKK7t0TaRVQ

**Comment by httpnotonly** [comment]
GOT IT

That's the right firmware, which is working with your plugin
https://github.com/SpacehuhnTech/esp8266_deauther/releases/download/2.6.1/esp8266_deauther_2.6.1_DSTIKE_NODEMCU_07.bin

haha  😈 
thanks, it's working

**Comment by httpnotonly** [comment]
I worked for 2 mins, than crashed in null pointer.. :)

**Comment by Timmotools** [comment]
Yeah the code is not perfect. I really cannot program in C. Learning as I go along. There's a big bug that if you send commands at the same time that data is being sent from the esp it will cause this error. 

Pretty much any of those bin files on their download work fine.  Just grab the one closest to your ESP board. 

I have also uploaded the unlshd-020 version of the fap. Turns out there's been no commits to unleashed since yesterday when I last updated my flipper.

**Comment by Timmotools** [comment]
I was playing with it today and it turns out that if you don't select "view log" before selecting the other options then it crashes most of the time. I believe I have now fixed this. Please download the latest FAP release and test. Please let me know if you have any issues.
